### PR TITLE
fix(api): Default pinned searches to have a name. Make sure correct value is returned after creation. (SEN-477)

### PR DESCRIPTION
**Pull request by wedamija**
We need to have a default name for pinned searches, just adding this in. Also making sure that the
pinned endpoint returns the correct search after pinning, since in some cases we'll be pinning a
global/org search.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_pinned_searches.py [hunk 1]
```diff
@@ -13,6 +14,9 @@
 from sentry.models.search_common import SearchType
 
 
+PINNED_SEARCH_NAME = 'My Pinned Search'
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_pinned_searches.py]: Should this be translated?

File: src/sentry/api/endpoints/organization_pinned_searches.py [hunk 2]
```diff
@@ -44,6 +49,20 @@ def put(self, request, organization):
...
                 type=result['type'],
             )
+            try:
+                # If we pinned an existing search, return the details about that
+                # search.
+                existing_search = SavedSearch.objects.filter(
+                    Q(organization=organization, owner__isnull=True) | Q(is_global=True),
+                    type=result['type'],
+                    query=result['query'],
+                )[:1].get()
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_pinned_searches.py]: Do we end up with two savedsearch rows with the same query/type and different owner values? Does the GET endpoint need to de-duplicate the pin + org search?

File: src/sentry/api/endpoints/organization_pinned_searches.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/api/endpoints/organization_pinned_searches.py]: No idea, do we translate the global search names?

File: src/sentry/api/endpoints/organization_pinned_searches.py (hunk 2 above)
Comment (wedamija) [reply src/sentry/api/endpoints/organization_pinned_searches.py]: Yeah, the list saved search endpoint already handles this.

File: src/sentry/api/endpoints/organization_pinned_searches.py (hunk 1 above)
Comment (markstory) [reply src/sentry/api/endpoints/organization_pinned_searches.py]: Probably not because we jam them in the database.

**Review by markstory — APPROVED** [review]

**Pull request merged**